### PR TITLE
fix depth for return for asm-wasm-builder

### PR DESCRIPTION
**Pull request by aseemgarg89**
Also fixed test-run-wasm-module

### CODE REVIEW

File: src/wasm/asm-wasm-builder.cc [hunk 1]
```diff
@@ -524,6 +527,7 @@ class AsmWasmBuilderImpl : public AstVisitor {
   Isolate* isolate_;
   Zone* zone_;
   TypeCache const& cache_;
+  int depthInFunction_;
```
Comment (titzer) [inline src/wasm/asm-wasm-builder.cc]: Can we name this block_depth?

Comment (aseemgarg89) [inline src/wasm/asm-wasm-builder.cc]: I was thinking depthInFunction_ and a stack of depthInLoop_ (after I implement loops, break, continue). Can also do block_depth and block_depth_in_loop.
Btw, what is the convention around using camel case vs snake case?

Comment (aseemgarg89) [inline src/wasm/asm-wasm-builder.cc]: done. Will use loop_depth_ for loops

Comment (binji) [inline src/wasm/asm-wasm-builder.cc]: @aseemgarg89 for naming conventions, see: https://www.chromium.org/developers/coding-style#Naming and https://google.github.io/styleguide/cppguide.html#Naming. Basically, snake_case for variable names, CamelCase for function names and class names.

Comment (titzer) [inline src/wasm/asm-wasm-builder.cc]: On Wed, Nov 4, 2015 at 3:27 PM, aseemgarg89 <notifications@github.com>
wrote:

> In src/wasm/asm-wasm-builder.cc
> <https://github.com/WebAssembly/v8-native-prototype/pull/50#discussion_r43956657>
> :
>
> > @@ -524,6 +527,7 @@ class AsmWasmBuilderImpl : public AstVisitor {
> >    Isolate* isolate_;
> >    Zone* zone_;
> >    TypeCache const& cache_;
> > +  int depthInFunction_;
>
> I was thinking depthInFunction_ and a stack of depthInLoop_ (after I
> implement loops, break, continue). Can also do block_depth and
> block_depth_in_loop.
>
You will probably need to keep a separate stack of loops to know exactly
which loop a JS continue or break statement targets.

**Pull request merged**